Mathlib/Analysis/NormedSpace/PiTensorProduct/InjectiveSeminorm.lean

```diff
@@ -0,0 +1,460 @@
+    ContinuousLinearMap.compContinuousMultilinearMap_coe, Function.comp_apply, tprodL_toFun]
+
+@[simp]
+theorem liftIsometry_tprodL :
+    liftIsometry 𝕜 E _ (tprodL 𝕜) = ContinuousLinearMap.id 𝕜 (⨂[𝕜] i, E i) := by
+  ext _
+  simp only [liftIsometry_apply_apply, tprodL_coe, lift_tprod, LinearMap.id_coe, id_eq,
+    ContinuousLinearMap.coe_id']
+
+end seminorm
+
+section map
+
+variable {E' E'' : ι → Type*}
+variable [∀ i, SeminormedAddCommGroup (E' i)] [∀ i, NormedSpace 𝕜 (E' i)]
+variable [∀ i, SeminormedAddCommGroup (E'' i)] [∀ i, NormedSpace 𝕜 (E'' i)]
+variable (g : Π i, E' i →L[𝕜] E'' i) (f : Π i, E i →L[𝕜] E' i)
+
+/--
+Let `Eᵢ` and `E'ᵢ` be two families of normed `𝕜`-vector spaces.
+Let `f` be a family of continuous `𝕜`-linear maps between `Eᵢ` and `E'ᵢ`, i.e.
+`f : Πᵢ Eᵢ →L[𝕜] E'ᵢ`, then there is an induced continuous linear map
+`⨂ᵢ Eᵢ → ⨂ᵢ E'ᵢ` by `⨂ aᵢ ↦ ⨂ fᵢ aᵢ`.
+-/
+noncomputable def mapL : (⨂[𝕜] i, E i) →L[𝕜] ⨂[𝕜] i, E' i :=
+  liftIsometry 𝕜 E _ <| (tprodL 𝕜).compContinuousLinearMap f
+
+@[simp]
+theorem mapL_coe : (mapL f).toLinearMap = map (fun i ↦ (f i).toLinearMap) := by
+  ext
+  simp only [mapL, LinearMap.compMultilinearMap_apply, ContinuousLinearMap.coe_coe,
+    liftIsometry_apply_apply, lift.tprod, ContinuousMultilinearMap.coe_coe,
+    ContinuousMultilinearMap.compContinuousLinearMap_apply, tprodL_toFun, map_tprod]
+
+@[simp]
+theorem mapL_apply (x : ⨂[𝕜] i, E i) : mapL f x = map (fun i ↦ (f i).toLinearMap) x := by
+  induction' x using PiTensorProduct.induction_on with _ _ _ _ hx hy
+  · simp only [mapL, map_smul, liftIsometry_apply_apply, lift.tprod,
+    ContinuousMultilinearMap.coe_coe, ContinuousMultilinearMap.compContinuousLinearMap_apply,
+    tprodL_toFun, map_tprod, ContinuousLinearMap.coe_coe]
+  · simp only [map_add, hx, hy]
+
+/-- Given submodules `pᵢ ⊆ Eᵢ`, this is the natural map: `⨂[𝕜] i, pᵢ → ⨂[𝕜] i, Eᵢ`.
+This is the continuous version of `PiTensorProduct.mapIncl`.
+-/
+@[simp]
+noncomputable def mapLIncl (p : Π i, Submodule 𝕜 (E i)) : (⨂[𝕜] i, p i) →L[𝕜] ⨂[𝕜] i, E i :=
+  mapL fun (i : ι) ↦ (p i).subtypeL
+
+theorem mapL_comp : mapL (fun (i : ι) ↦ g i ∘L f i) = mapL g ∘L mapL f := by
+  apply ContinuousLinearMap.coe_injective
+  ext
+  simp only [mapL_coe, ContinuousLinearMap.coe_comp, LinearMap.compMultilinearMap_apply, map_tprod,
+    LinearMap.coe_comp, ContinuousLinearMap.coe_coe, Function.comp_apply]
+
+theorem liftIsometry_comp_mapL (h : ContinuousMultilinearMap 𝕜 E' F) :
+    liftIsometry 𝕜 E' F h ∘L mapL f = liftIsometry 𝕜 E F (h.compContinuousLinearMap f) := by
+  apply ContinuousLinearMap.coe_injective
+  ext
+  simp only [ContinuousLinearMap.coe_comp, mapL_coe, LinearMap.compMultilinearMap_apply,
+    LinearMap.coe_comp, ContinuousLinearMap.coe_coe, Function.comp_apply, map_tprod,
+    liftIsometry_apply_apply, lift.tprod, ContinuousMultilinearMap.coe_coe,
+    ContinuousMultilinearMap.compContinuousLinearMap_apply]
+
+attribute [local ext high] ext
```

Comment:
Is this line really useful?

Reply:
CI is happy, so I guess the line was not useful.